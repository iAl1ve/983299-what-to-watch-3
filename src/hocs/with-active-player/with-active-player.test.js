import React from "react";
import renderer from "react-test-renderer";
import {oneOfType, arrayOf, node} from "prop-types";
import withActivePlayer from './with-active-player.jsx';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: oneOfType([
    arrayOf(node),
    node
  ]),
};

const MockComponentWrapped = withActivePlayer(MockComponent);

it(`withActivePlayer renders correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped/>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
