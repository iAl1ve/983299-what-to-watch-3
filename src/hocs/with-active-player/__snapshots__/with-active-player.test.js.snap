// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActivePlayer renders correctly 1`] = `<div />`;
